xarray/core/dataarray.py

```diff
@@ -1086,6 +1086,35 @@ def unstack(self, dim):
         ds = self._to_temp_dataset().unstack(dim)
         return self._from_temp_dataset(ds)
 
+    def unstack_cat(self, dim, level=0):
+        """Unstack DataArray expanding to Dataset along a given level
+
+        Parameters
+        ----------
+        dim : str
+            Name of existing dimension to unstack
+        level : int
+            Index of level to expand to dataset along
+
+        Returns
+        -------
+        unstacked: Dataset
+
```

Comment:
An example here would also be helpful, as well as a "See also" reference to `stack_cat`.

Reply:
@nbren12 - we could still use an example here.

Reply:
Resolved below